src/units/abilities.cpp

```diff
@@ -898,6 +898,61 @@ std::vector<std::pair<t_string, t_string>> attack_type::special_tooltips(
 	return res;
 }
 
+std::vector<std::pair<t_string, t_string>> attack_type::abilities_special_tooltips(
+	boost::dynamic_bitset<>* active_list) const
+{
+	std::vector<std::pair<t_string, t_string>> res;
+	if(active_list) {
+		active_list->clear();
+	}
+	std::set<std::string> checking_name;
+	if(self_) {
+		for(const auto [key, cfg] : self_->abilities().all_children_view()){
+			if(!active_list || check_self_abilities_impl(shared_from_this(), other_attack_, cfg, self_, self_loc_, AFFECT_SELF, key, false)) {
+				std::string name = cfg["name_affected"];
```

Comment:
could be `const`? same next line.
```suggestion
				const std::string name = cfg["name_affected"];
```